addons/sale/models/sale.py

```diff
@@ -695,6 +695,12 @@ def action_confirm(self):
         self._action_confirm()
         if self.env['ir.config_parameter'].sudo().get_param('sale.auto_done_setting'):
             self.action_done()
+
+        tx = self.sudo().transaction_ids.get_last_transaction()
```

Comment:
I'd move this to `action_done`. When done, you are not supposed to do anything on the SO anymore.